### PR TITLE
Sema: Remove @noreturn attribute support

### PR DESCRIPTION
**Pull request by slavapestov**
Swift 5 removed Swift 3 mode, but kept some diagnostic code around to emit
a fixit. It's time to remove that now.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: include/swift/AST/Attr.def [hunk 1]
```diff
@@ -136,9 +135,7 @@ CONTEXTUAL_SIMPLE_DECL_ATTR(optional, Optional,
 SIMPLE_DECL_ATTR(dynamicCallable, DynamicCallable,
   OnNominalType,
   6)
-SIMPLE_DECL_ATTR(noreturn, NoReturn,
-  OnFunc | OnAccessor,
-  7)
+// NOTE: 6 is unused
```
Comment (aroben) [inline include/swift/AST/Attr.def]: Looks like 7 is unused? Or maybe I’m misreading.

Comment (slavapestov) [reply include/swift/AST/Attr.def]: You’re absolutely right.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**